src/commits.js

```diff
@@ -35,7 +35,8 @@ const parseCommits = (string, options = {}) => {
     .split(COMMIT_SEPARATOR)
     .slice(1)
     .map(commit => parseCommit(commit, options))
-    .filter(commit => filterCommit(commit, options))
+    .filter(commit => excludeCommit(commit, options))
+    .filter(commit => includeCommit(commit, options))
```

Comment:
what happens if include and exclude have overlaps? that seems like it should be an error rather than just arbitrarily deciding which one takes precedence.

Reply:
well I thought about this for while before making this PR, and what I concluded after a bit of tests was that it doesn't really matter which takes precedence, the result will stay the same. 
or at least from what I tested trying to come up with something that would break it and couldn't.
I'd say they can coexist without any issue, you can have both a blacklist and whitelist at the same time, and you are the one responsible for making it work the way you like to autogen your changeling, but these are simply options you have to work with.
so Im happy to apply necessary changes if you can give me an example that this might break or not work as it should

Reply:
I think the order matters if there's a file that's covered by both patterns.

Reply:
well maybe add a new flag so the precedence can be chosen?

Reply:
is there a use case to provide both? perhaps the question could be deferred by making the two flags mutually exclusive.

Reply:
well I assume there can be a case that both can be used, but probably that would be less than 5% of cases. but if you think making thme exclusive is a safer option, we can go with that

Reply:
It can always be allowed later without being semver-major, but it can't be disallowed later without being semver-major.